mineinabyss-features/src/main/kotlin/com/mineinabyss/displayLocker/DisplayLockerFeature.kt

```diff
@@ -0,0 +1,84 @@
+package com.mineinabyss.displayLocker
+
+import com.mineinabyss.components.armorstandlock.LockArmorStand
+import com.mineinabyss.components.playerData
+import com.mineinabyss.geary.papermc.access.toGeary
+import com.mineinabyss.geary.papermc.store.encodeComponentsTo
+import com.mineinabyss.idofront.commands.arguments.stringArg
+import com.mineinabyss.idofront.commands.extensions.actions.playerAction
+import com.mineinabyss.idofront.entities.toPlayer
+import com.mineinabyss.idofront.messaging.broadcast
+import com.mineinabyss.idofront.messaging.error
+import com.mineinabyss.idofront.messaging.info
+import com.mineinabyss.idofront.messaging.success
+import com.mineinabyss.idofront.plugin.registerEvents
+import com.mineinabyss.mineinabyss.core.AbyssFeature
+import com.mineinabyss.mineinabyss.core.MineInAbyssPlugin
+import com.mineinabyss.mineinabyss.core.commands
+import kotlinx.serialization.SerialName
+import net.minecraft.util.DirectoryLock.LockException
+import org.bukkit.Bukkit
+import org.bukkit.entity.Player
+
+
+@kotlinx.serialization.Serializable
+@SerialName("lock_display_item")
+class DisplayLockerFeature: AbyssFeature {
+    override fun MineInAbyssPlugin.enableFeature() {
+        registerEvents(DisplayLockerListener())
+
+        commands {
+            mineinabyss {
+                "lock"(desc="display locker commands") {
+                    val player = sender as Player
+                    val entity = player.playerData.recentRightclickedEntity
+                    val locked = entity?.toGeary()?.get<LockArmorStand>()
+
+                    "toggle"(desc="toggles locking on display item") playerAction@{
+                        locked?.lockState = !locked?.lockState!!
```

Comment:
if entity is null here this will throw an error. hence ithink i kept it within each cmd so i could return in the elvis statement
`togeary?.get<>() ?: return@playerAction`. Ensures there is an entity

Reply:
fixed in [fc06d0a](https://github.com/MineInAbyss/MineInAbyss/pull/159/commits/fc06d0a70deac3da2756e22323381d41573f1935)